### PR TITLE
Add possibility to check signature

### PR DESCRIPTION
**Pull request by dhodovsk**
Mostly copy and paste from code written by lzachar.

Additional requested functionality not covered in this PR:
 - auto detection of allowed keys
 - print origin (repository) of not properly signed packages in logs

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/Rxi7g](https://da.gd/Rxi7g)__
* Comment: __See details for logs__

File: conu/utils/rpms.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    used_key = key_match.group(1)
+    if used_key in allowed_keys:
+        return True
+    logger.error("Wrong key for '{0}' ({1})".format(name, used_key))
+    return False
+
+
+def check_signatures(pkg_list, allowed_keys):
+    """
+    Go through list of packages with signatures and check if or all properly signed
```
Comment (TomasTomecek) [inline conu/utils/rpms.py]: docs nit: `if all are properly signed`

File: conu/backend/docker/image.py [hunk 2]
```diff
@@ -232,6 +233,31 @@ def callback():
...
+         * parsing of rpm fails
+         * there are packages in image that are not signed with one of allowed keys
+
+        :param allowed_keys: list of allowed keys
+        :return: bool
+        """
+
+        if not allowed_keys or not isinstance(allowed_keys, list):
+            raise ConuException("allowed_keys must be a list")
+        self.pull()
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: I would probably not pull the image implicitly here

File: conu/backend/docker/image.py [hunk 3]
```diff
@@ -232,6 +233,31 @@ def callback():
...
+        drb = DockerRunBuilder(command=['rpm', '-qa', '--qf', '%{name} %{SIGPGP:pgpsig}\n'])
+        cont = self.run_via_binary(drb)
+        out = cont.logs_unicode()[:-1].split('\n')
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: even logs can actually throw an exc so I would put it inside the try/finally block

File: conu/utils/rpms.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+from conu.exceptions import PackageSignatureException
+
+NONE_KEY = '(none)'
+KEY = r'.*Key ID ([a-h0-9]+)'
+no_key_pkgs = ['gpg-pubkey']
+
+logger = logging.getLogger(__name__)
+
+
+def process_rpm_ql_line(line_str, allowed_keys):
```
Comment (TomasTomecek) [inline conu/utils/rpms.py]: I really like the implementation of this method.

File: conu/backend/docker/image.py [hunk 5]
```diff
@@ -232,6 +233,31 @@ def callback():
 
         return DockerContainer(self, container_id, popen_instance=popen_instance, name=container_name)
 
+    def has_pkgs_signed_with(self, allowed_keys):
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: maybe add to readme that we conu can do this?

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/7Uha3](https://da.gd/7Uha3)__
* Comment: __See details for logs__

**Pull request merged**